datafusion/expr/src/logical_plan/ddl.rs

```diff
@@ -303,3 +321,66 @@ pub struct DropCatalogSchema {
     /// Dummy schema
     pub schema: DFSchemaRef,
 }
+
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct CreateFunction {
+    // TODO: There is open question should we expose sqlparser types or redefine them here?
+    //       At the moment it make more sense to expose sqlparser types and leave
+    //       user to convert them as needed
+    pub or_replace: bool,
+    pub temporary: bool,
+    pub name: String,
+    pub args: Option<Vec<OperateFunctionArg>>,
+    pub return_type: Option<DataType>,
+    // TODO: move this to new struct here
+    pub params: CreateFunctionBody,
+    //pub body: String,
+    /// Dummy schema
+    pub schema: DFSchemaRef,
+}
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct OperateFunctionArg {
+    // it is not really supported so no need to have it here
+    // currently
```

Comment:
I think it is fine to leave some of these fields as "TODO" items and if someone needs them they can add them as a follow on PR
```suggestion
    // TODO: figure out how to support mode
```